internal/graphql/model/helpers.go

```diff
@@ -114,3 +140,182 @@ func applyYaml(client *mesherykube.Client, delete bool, file string) error {
 
 	return nil
 }
+
+// installUsingHelm is for installing helm dependencies. We need this because
+// meshery operator and controllers don't have published separate charts but
+// exist as subcharts for meshery's chart.
+// We plan to have separate charts for these components once we wish to offer
+// users the control over which version of helm want to use
+func installUsingHelm(client *mesherykube.Client, delete bool) error {
+	releaseVersion := viper.GetString("BUILD")
+	if releaseVersion == "" || releaseVersion == "Not Set" || releaseVersion == "edge-latest" {
+		latestReleaseData, err := handlers.CheckLatestVersion("")
+		if err != nil {
+			releaseVersion = latestReleaseData.Current
+		}
+	}
+
+	var (
+		act            = mesherykube.INSTALL
+		chart          = "meshery-operator"
+		overrideValues map[string]interface{}
+		err            error
+	)
+
+	// if meshery server is in cluster, we upgrade the release
+	if viper.GetString("KUBERNETES_SERVICE_HOST") != "" {
+		act = mesherykube.UPGRADE
+		chart = "meshery"
+	} else if delete {
+		act = mesherykube.UNINSTALL
+	}
+
+	if delete {
+		// Delete the CR instances for brokers and meshsyncs
+		// this needs to be executed before deleting the helm release, or the CR instances cannot be found for some reason
+		if err := DeleteCR(brokerResourceName, brokerInstanceName, client); err != nil {
+			return errors.Wrap(err, "cannot delete CR "+brokerInstanceName)
+		}
+		if err := DeleteCR(meshsyncResourceName, meshsyncInstanceName, client); err != nil {
+			return errors.Wrap(err, "cannot delete CR "+meshsyncInstanceName)
+		}
+	}
+
+	overrideValues, err = SetOverrideValues(delete)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Installing operator using chart: %s", chart)
+
+	err = client.ApplyHelmChart(mesherykube.ApplyHelmChartConfig{
+		Namespace: "meshery",
+		ChartLocation: mesherykube.HelmChartLocation{
+			Repository: chartRepo,
+			Chart:      chart,
+			Version:    releaseVersion,
+		},
+		CreateNamespace: true,
+		Action:          act,
+		OverrideValues:  overrideValues,
+	})
+	if err != nil {
+		return err
+	}
+
+	if delete {
+		// Delete the CRDs for brokers and meshsyncs
+		// These need to be deleted after deleting the chart
+		if err = DeleteCRD(brokderCRDName); err != nil {
+			return errors.Wrap(err, "cannot delete CRD "+brokderCRDName)
+		}
+
+		if err = DeleteCRD(meshsyncCRDName); err != nil {
+			return errors.Wrap(err, "cannot delete CRD "+meshsyncCRDName)
+		}
+	}
+
+	return nil
+}
+
+// SetOverrideValues detects the currently insalled adapters and sets appropriate
+// overrides so as to not uninstall them. It also sets overrides for operator to
+// false or true if it is a delete operation
+func SetOverrideValues(delete bool) (map[string]interface{}, error) {
+	installedAdapters := make([]string, 0)
+	adapters := helpers.NewAdaptersTracker(viper.GetStringSlice("ADAPTER_URLS")).GetAdapters(context.TODO())
```

Comment:
@utkarsh-pro @Revolyssup I want to check what all adapters are installed and apply overrides accordingly so that upgrading the helm release doesn't uninstall any of them.
`GetAdapters()` should return all the data about adapters which can then be used to check which one is installed or not, right? Basically, this is the one behind `/api/system/adapters` endpoint's response.
However, it doesn't seem to have all the data. Do you know why?